### PR TITLE
cleanup around AdamOptimizer

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)
Small changes to make it potentially easier in the future to inject a new Optimizer type.
@ervteng Please let me know if this is going to conflict with torch changes; we can probably delay.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [x] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/optimizer/tf_optimizer.py [hunk 1]
```diff
@@ -140,8 +140,9 @@ def create_reward_signals(
                 self.reward_signals[reward_signal.value].update_dict
             )
 
+    @classmethod
     def create_optimizer_op(
-        self, learning_rate: tf.Tensor, name: str = "Adam"
+        cls, learning_rate: tf.Tensor, name: str = "Adam"
```
Comment (chriselion) [inline ml-agents/mlagents/trainers/optimizer/tf_optimizer.py:145]: This didn't need the actual instance, but can still be overridden by derived classes.

File: ml-agents/mlagents/trainers/ppo/optimizer.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def __init__(self, policy: TFPolicy, trainer_params: TrainerSettings):
 
                 self.stream_names = list(self.reward_signals.keys())
 
-                self.tf_optimizer: Optional[tf.train.AdamOptimizer] = None
+                self.tf_optimizer_op: Optional[tf.train.Optimizer] = None
```
Comment (chriselion) [inline ml-agents/mlagents/trainers/ppo/optimizer.py:43]: Slightly more generic type, and rename to make clear that it's not a TFOptimizer instance.

File: ml-agents/mlagents/trainers/ppo/trainer.py [hunk 3]
```diff
@@ -208,6 +208,9 @@ def create_policy(
 
         return policy
 
+    def create_ppo_optimizer(self) -> PPOOptimizer:
```
Comment (chriselion) [inline ml-agents/mlagents/trainers/ppo/trainer.py:211]: Made these so we can wrap the TFOptimizer impls more easily. I briefly tried adding
```
    @abc.abstractmethod
    def create_tf_optimizer(self) -> TFOptimizer:
```
in RLTrainer but I think at least SACTrainer needs the specific type. Do you agree, or should I try harder to push this higher in the hierarchy?

**Comment by chriselion** [comment]
@ervteng Sorry, mind taking another look? I pushed some other small changes after the initial approval.

File: ml-agents/mlagents/trainers/optimizer/tf_optimizer.py (hunk 1 above)
Comment (anupambhatnagar) [reply ml-agents/mlagents/trainers/optimizer/tf_optimizer.py:145]: 👍

File: ml-agents/mlagents/trainers/ppo/optimizer.py (hunk 2 above)
Comment (anupambhatnagar) [reply ml-agents/mlagents/trainers/ppo/optimizer.py:43]: 👍

File: ml-agents/mlagents/trainers/ppo/trainer.py (hunk 3 above)
Comment (anupambhatnagar) [reply ml-agents/mlagents/trainers/ppo/trainer.py:211]: LGTM

**Pull request merged**